modules/Components/src/client/ACASFormFields.coffee

```diff
@@ -183,24 +183,73 @@ class window.ACASFormLSLabelFieldController extends ACASFormAbstractFieldControl
 		@userInputEvent = true
 		value = UtilityFunctions::getTrimmedInput(@$('input'))
 
-		if @isValid(value)
-			if value != ""
-				@getModel().set
-					labelText: value
-					ignored: false
-			else
-				@setEmptyValue()
-		super()
+		# check
+		@isValid value, =>
+			super()
 
-	isValid: (value) ->
+	isValid: (value, callback) =>
+		@checkUnique value, (isValid) =>
+			if isValid
+				@clearError()
+				if value != ""
+					@getModel().set
+						labelText: value
+						ignored: false
+				else
+					@setEmptyValue()
+			callback()
+
+	checkUnique: (value, callback) ->
+		if @getModel().get("validationRegex")?
```

Comment:
I get the logical flow, but it's strange to me that the "validationRegex" check is under a function called "checkUnique".

Reply:
Yeah definitely not clear and easy to read or easily reusable.  I just pushed an update which I think is much better.